### PR TITLE
Remaining tests for the Python Image Segmenter Task Library

### PR DESCRIPTION
**Pull request by kinaryml**
- Fixed a `SegmentationOptions` initialization issue in `ImageSegmenterOptions`
- Added tests for category mask and confidence mask along with some additional tests

### CODE REVIEW

File: tensorflow_lite_support/python/test/task/vision/image_segmenter_test.py [hunk 1]
```diff
@@ -196,6 +231,80 @@ def test_segment_model(self, model_file_type, expected_colored_labels):
...
+    inconsistent_pixels = 0
+
+    for index in range(num_pixels):
+      inconsistent_pixels += (
+          result_pixels[index] * _MASK_MAGNIFICATION_FACTOR !=
+          ground_truth_pixels[index])
+
+    self.assertLessEqual(inconsistent_pixels / num_pixels,
+                         _MATCH_PIXELS_THRESHOLD,
+                         "Segmentation mask value must be the same size as ground truth.")
```
Comment (khanhlvg) [inline tensorflow_lite_support/python/test/task/vision/image_segmenter_test.py:265]: The error message is incorrect here.

Besides, you should also assert if the segmentation mask size is the same as the ground truth mask size before the for loop.

**Review by khanhlvg — CHANGES_REQUESTED** [review]
I've left some comments. PTAL. Thanks!

File: tensorflow_lite_support/python/test/task/vision/image_segmenter_test.py [hunk 2]
```diff
@@ -196,6 +231,85 @@ def test_segment_model(self, model_file_type, expected_colored_labels):
...
+    result_pixels = np.array(bytearray(segmentation.category_mask))
+
+    # Loads ground truth segmentation file.
+    gt_segmentation = tensor_image.TensorImage.create_from_file(
+      self.test_seg_path)
+    gt_segmentation_array = gt_segmentation.buffer
+    gt_segmentation_shape = gt_segmentation_array.shape
+    num_pixels = gt_segmentation_shape[0] * gt_segmentation_shape[1]
+    ground_truth_pixels = gt_segmentation_array.flatten()
+
```
Comment (kinaryml) [inline tensorflow_lite_support/python/test/task/vision/image_segmenter_test.py:255]: @khanhlvg I've added a check here before running the loop and also updated the error message to a more appropriate one. Let me know if you'd like to revise it.

Comment (khanhlvg) [reply tensorflow_lite_support/python/test/task/vision/image_segmenter_test.py:255]: LGTM. Thanks!

File: tensorflow_lite_support/python/test/task/vision/image_segmenter_test.py [hunk 3]
```diff
@@ -196,6 +231,85 @@ def test_segment_model(self, model_file_type, expected_colored_labels):
...
+                     'categories.')
+
+    # Gather the confidence masks in a single array `confidence_mask_array`.
+    confidence_mask_array = np.array([
+      confidence_masks[index].value for index in range(len(confidence_masks))])
+
+    # Compute the category mask from the created confidence mask.
+    calculated_category_mask = np.argmax(confidence_mask_array, axis=0)
+    self.assertListEqual(calculated_category_mask.tolist(),
+                         category_mask.tolist())
```
Comment (khanhlvg) [inline tensorflow_lite_support/python/test/task/vision/image_segmenter_test.py:311]: Please add a message here.

File: tensorflow_lite_support/python/test/task/vision/image_segmenter_test.py [hunk 4]
```diff
@@ -196,6 +231,85 @@ def test_segment_model(self, model_file_type, expected_colored_labels):
...
+    # Performs image segmentation on the input again.
+    segmentation = segmenter.segment(image).segmentation[0]
+
+    # Gets the list of confidence masks and colored_labels.
+    confidence_masks = segmentation.confidence_masks.confidence_mask
+    colored_labels = segmentation.colored_labels
+
+    # Check if confidence mask shape is correct.
+    self.assertEqual(len(confidence_masks), len(colored_labels),
+                     'Number of confidence masks must match with number of '
```
Comment (khanhlvg) [inline tensorflow_lite_support/python/test/task/vision/image_segmenter_test.py:301]: Please don't mix single quote and double quote in the same file. Let's unify to double quote then.

**Review by khanhlvg — APPROVED** [review]

**Pull request merged**